instances/etcd/main.tf

```diff
@@ -29,3 +29,18 @@ resource "oci_core_instance" "TFInstanceEtcd" {
     command = "sleep 10"
   }
 }
+
+resource "oci_core_volume" "TFVolumeInstanceEtcd" {
+  count               = "${var.etcd_iscsi_volume_create ? var.count : 0}"
+  availability_domain = "${var.availability_domain}"
+  compartment_id      = "${var.compartment_ocid}"
+  size_in_gbs         = "${var.etcd_iscsi_volume_size}"
+}
+
+resource "oci_core_volume_attachment" "TFVolumeAttachmentInstanceEtcd" {
+  count           = "${var.etcd_iscsi_volume_create ? var.count : 0}"
+  attachment_type = "iscsi"
+  compartment_id  = "${var.compartment_ocid}"
+  instance_id     = "${oci_core_instance.TFInstanceEtcd.id}"
```

Comment:
Sorry for the delay. ,We just found an issue with the worker code. I think the code _should_ be:

```
  instance_id     = "${oci_core_instance.TFInstanceEtcd.*.id[count.index]}"
  volume_id       = "${element(oci_core_volume.TFInstanceEtcd.*.id, count.index)}"
  depends_on      = ["oci_core_instance.TFInstanceEtcd", "oci_core_volume.TFVolumeInstanceEtcd"]
```

Reply:
Just made the change for worker nodes [here](https://github.com/oracle/terraform-kubernetes-installer/commit/cbe53f329c8bfcf5e80195deac5ad8efd63cc0b2).